regtest/tools/rt-make-random/main.cpp

```diff
@@ -0,0 +1,78 @@
+#include "plumed/tools/LatticeReduction.h"
+#include "plumed/tools/Stopwatch.h"
+#include "plumed/tools/Random.h"
+#include <fstream>
+#include <iostream>
+
+using namespace PLMD;
+
+int testConstructor(std::ofstream &ofs) {
+  ofs << "testConstructor" << std::endl;
+  new Random("random");
+  return 0;
+}
+
+int testRandU01(std::ofstream &ofs) {
+  ofs << "testRandU01" << std::endl;
+  Random* random = new Random("random");
```

Comment:
Oh man of course, this is a great catch. Is there any automated tooling in C++ world that can detect brain-farts like those? ;-)

Reply:
@jandom I am not sure... But the general recommendation (starting with C++11) is to always use smart pointers (`unique_ptr` and `shared_ptr`) and never use `new` and `delete` operators (or use `new` just to immediately build a `unique_ptr`). In this manner leaks should be prevented. So, the tool that I suggest is grep: `git grep " new "` :-)

C++ is continuously growing, so you always find a mixture of old and new constructs, but I would really recommend to start with C++11 since it has major improvements (including smart pointers).

Reply:
👏

Reply:
Hahah, this was awesome. You know what i'd love? And it's really not grep! What would be awesome, is a code analysis tool that makes this easy for me – like you configure it to allow only features from C++11 and above and discard everything else. "Oh this old school thing you're trying to do? Yeah it compiles but you really shouldn't do it"

Reply:
If you find such a tool, let me know... I am seriously using grep to detect these. Some time ago I used it to go through the whole set of default modules and replace all the pointers with unique_ptrs.